### DISCUSSION TITLE
Issue Reading from I2C Register

### DISCUSSION THREAD

**Discussion by sageduggal** [micropython/micropython#10563] [Using MicroPython] [answered]
Hello there,

I am attempting to write to a specific register, however when I check the register after writing, I am getting an odd output.

```py
i2c.writeto_mem(i2caddress, PWR_MGMT_1, bytearray([0b00000000]))
print(i2c.readfrom_mem(i2caddress, PWR_MGMT_1, 1))
i2c.writeto_mem(i2caddress, PWR_MGMT_1, bytearray([0b00101000]))
print(i2c.readfrom_mem(i2caddress, PWR_MGMT_1, 1))
```

The first print output is correct:
```py
b'\x00'
```

The second one, however, I do not understand:
```py
b'('
```

Has anyone experienced this before? I must be doing something wrong but not sure what is going on. Thank you in advance.

**Comment by peterhinch** [comment, marked as answer]
The '(' character corresponds to 0x28 (try `hex(ord(b'('))`). Is this an unreasonable response?

**Reply by sageduggal** [reply]
Amazing, thank you! I was pulling my hair out trying to figure out what was happening. Thanks again.

**Reply by peterhinch** [reply]
The way that Python prints `bytes` and `bytarray` objects is a frequent cause of confusion. It's worth spending a bit of time getting really familiar with it so that this makes sense:
```python
>>> bytearray((1,9,10,13,64))
bytearray(b'\x01\t\n\r@')
```
